components/script/dom/node.rs

```diff
@@ -168,6 +168,23 @@ pub struct Node {
     layout_data: DomRefCell<Option<Box<GenericLayoutData>>>,
 }
 
+impl fmt::Debug for Node {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if matches!(self.type_id(), NodeTypeId::Element(_)) {
+            let el = self.downcast::<Element>().unwrap();
+            el.fmt(f)
+        } else {
+            write!(f, "[Node({:?})]", self.type_id())
+        }
+    }
+}
+
+impl fmt::Debug for DomRoot<Node> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        (**self).fmt(f)
+    }
+}
+
```

Comment:
This turned out to be handy for debugging... maybe there's a reason this wasn't there before though?